packages/ui-components/src/components/AppShellProvider/AppShellProvider.component.tsx

```diff
@@ -0,0 +1,62 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Juno contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React from "react"
+
+import { StyleProvider } from "../StyleProvider"
+import { ShadowRoot } from "../ShadowRoot"
+import { PortalProvider } from "../PortalProvider"
+import { DEFAULT_THEME_NAME } from "../StyleProvider/StyleProvider.component"
+
+const Wrapper = ({ children, shadowRoot, shadowRootMode }: WrapperProps) => {
+  return shadowRoot ? (
+    <ShadowRoot mode={shadowRootMode}>
+      <>{children}</>
+    </ShadowRoot>
+  ) : (
+    children
+  )
+}
+
+/**
+ * This provider acts as a wrapper for Juno apps. It renders a StyleProvider and PortalProvider
+ */
+export const AppShellProvider = ({
+  shadowRoot = true,
+  shadowRootMode = "open",
+  stylesWrapper = "inline",
+  theme = DEFAULT_THEME_NAME,
+  children,
+}: AppShellProviderProps) => {
+  return (
+    <Wrapper shadowRoot={shadowRoot} shadowRootMode={shadowRootMode}>
+      <StyleProvider theme={theme} stylesWrapper={shadowRoot ? "inline" : stylesWrapper}>
+        <PortalProvider>{children}</PortalProvider>
+      </StyleProvider>
+    </Wrapper>
+  )
+}
+
+export type AppShellStyleWrapper = "head" | "inline"
+
+interface WrapperProps {
```

Comment:
little comment from my side you can use it as absed props for AppShellProviderProps and extends it removing them from AppShellProviderProps

Reply:
Done